VERSIONING.md

```diff
@@ -0,0 +1,145 @@
+# Versioning
+
+Universal Crossplane (UXP) uses a semantic versioning system based on the
+upstream Crossplane versioning to identify releases. UXP embeds Crossplane
+itself in its Helm chart, so the major, minor, and patch version of any UXP
+release will always match the embedded Crossplane major, minor, and patch
+version. The following sections describe the UXP versioning strategy, including
+when branches are cut, dependency tags are incremented, and new versions are
+released.
+
+## Format
+
+The UXP version string is formatted as a semantic version with appended
+pre-release information:
+
+```
+v<major>.<minor>.<patch>-up.<components>
+```
+
+The `<major>`, `<minor>`, and `<patch>` segments each refer to the corresponding
+embedded Crossplane `<major>`, `<minor>`, and `<patch>` versions. The embedded
+Crossplane version may differ from upstream Crossplane (i.e. Upbound may release
+patch versions past the end of life date for a Crossplane branch), but if the
+specified version exists upstream, the contents will match exactly.
+
+The `<components>` segment is a numeric identifier describing the collective
+version of software components that surround Crossplane in a UXP release. The
+`-up.<components>` segment must be a [valid smever pre-release
+segment](https://semver.org/#spec-item-9). Pre-release information is used
+instead of [build metadata](https://semver.org/#spec-item-10) as pre-release
+precedence in honored in semantic versioning. The `<components>` numeric
+identifier only corresponds to the version of components for its preceding
+`<major>`, `<minor>`, `<patch>` versions. There is no relationship between
+`<components>` versions across `<major>`, `<minor>`, `<patch>` versions.
+
+An example version string for a UXP release that embeds Crossplane `v1.3.0`
+could be:
+
+```
+v1.3.0-up.2
+```
+
+Additional pre-release segments may be appended both before and after the
+`-up-<components>` segment. Any pre-release segments appended before will
+correspond to the embedded Crossplane version (e.g. `v1.4.0-rc.0-up.1`). Any
+pre-release segments appended after will correspond to a pre-release version of
+UXP itself (e.g. `v1.4.0-up.1-rc.0`). There should _never_ be both a Crossplane
+pre-release and UXP pre-release on the same version string (see [Guarantees
+section](#guarantees)).
```

Comment:
I don't disagree with this practice, or have any better ideas, but I am a little worried that it will be very easy to mix up an RC version of a UXP release (e.g. `v1.3.0-up.1-rc.0`) and a UXP release of an upstream RC version (e.g. `v1.3.0-rc.0-up.1`).

Reply:
One fairly gross way to clarify would be to use something like `uprc` to distinguish UXP release candidates. e.g. `v1.3.0-up.1-uprc.0`. 😬

Reply:
I am not strongly opposed to this and don't think it takes us any farther away from conventions than we already are. Happy to go either way on this -- @turkenh do you have thoughts?

Reply:
I am still wondering if we could completely drop off `-up.1` in UXP main as I commented before :)

Reply:
@turkenh IMO that would break our versioning format because we are essentially extending our semantic version specification. I would equate it with tagging a Crossplane version with `v1.2-rc.0` instead of `v1.2.0-rc.0` on `master` (though I do acknowledge this is not an exact parallel 🙂 ). Because these will only be RC versions anyway, let's start off with keeping the `-up.X` version for now and we can change course in the future if needed 👍🏻 

Also, I decided to add the `uprc` change 👍🏻